secret-manager/runtime/src/main/java/io/quarkiverse/googlecloudservices/secretmanager/runtime/config/SecretManagerConfigSourceProvider.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkiverse.googlecloudservices.secretmanager.runtime.config;
+
+import java.util.Collections;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+import com.google.cloud.ServiceOptions;
+
+import io.quarkiverse.googlecloudservices.common.GcpBootstrapConfiguration;
+
+public class SecretManagerConfigSourceProvider implements ConfigSourceProvider {
+
+    private final String projetId;
+
+    public SecretManagerConfigSourceProvider(GcpBootstrapConfiguration gcpBootstrapConfiguration) {
+        this.projetId = gcpBootstrapConfiguration.projectId;
+    }
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources(ClassLoader forClassLoader) {
+        String projectId = this.projetId;
```

Comment:
Instead of reassigning the variable, just check `if (this.projectId == null)`?

Reply:
Yes, it's not very neat.
I changed to the ternary operator.